### DISCUSSION TITLE
State contain only the actual one

### DISCUSSION THREAD

**Discussion by Mte90** [spatie/laravel-livewire-wizard#55] [Q&A]
So I have a webapp with various wizardcomponents and I see them in the navigation and also I am able to move to the other, save things and so on.

The issue is on `state` that if I execute just return only the state of the actual stepcomponent also on `all`.
To move to the next steps I have a function that use `nextStep` and I don't understand why the state are missing.

Reading the docs and other discussions I wasn't able to find any hints about it. Maybe the state are no available on `render` or on `boot`?

**Comment by Mte90** [comment]
In the meantime I will use the livewire events to keep working on my webapp.

**Reply by Mte90** [reply]
I am also wondering if it is possible to send a data to the next step, as I have implemented a way to set the step and that data with the URL so the user can restart from a specific point.

**Comment by ju5t** [comment]
I don't understand your initial question. Maybe someone else can help with that.

About starting at a specific step and passing data:
- https://spatie.be/docs/laravel-livewire-wizard/v1/usage/navigating-steps#content-start-at-a-specific-step
- https://spatie.be/docs/laravel-livewire-wizard/v1/usage/setting-initial-state

This is all you need for it.

**Reply by Mte90** [reply]
At the end I used a cookie to save that variable and share it to the next steps.
So if the states doesn't have that I can recover it in that way, also because in my case you need to be able to continue the wizard in a second moment so I generate url with a parameter to restart from here and state seems just readonly.

My issue just for documentation is that state doesn't contain data from the previous states also if the session is the same.

**Reply by ju5t** [reply]
Have you tried it like the below code sample? This should work. It's how we're fetching data from specific steps in our wizards.

```
$this->state()->forStep('livewire-alias');
```

Cookies shouldn't be necessary, but it makes sense for users to recover their session. The package doesn't have any opinion about it.
